message_ui.module

```diff
@@ -1,447 +1,245 @@
-}
+const MESSAGE_UI_DENY = TRUE;
 
 /**
- * Check if the user can create an instance for a message type.
+ * Implements hook_entity_base_field_info_alter.
  *
- * @param $type
- *  The message type for which the info shall be returned, or NULL to return an
- *  array with info about all types.
- * @param $account
- *  The user object or user uid.
- *
- * @return array|bool
- *  TRUE or FALSE for a specific message type or an array of the message types
- */
-function message_ui_user_can_create_message($type = NULL, $account = NULL) {
-  if (empty($account)) {
-    global $user;
-    $account = $user;
-  }
-
-  $account = entity_metadata_wrapper('user', $account)->value();
-  $types = message_ui_get_types();
-
-  // User have access to create any instances.
-  if (user_access('create any message instance', $account)) {
-    return TRUE;
-  }
-
-  // Check access for a specific message.
-  if ($type) {
-
-    // Didn't found that type.
-    if (!in_array($type, $types)) {
-      return;
-    }
-
-    if (user_access('create a ' . $type . ' message instance', $account)) {
-      return TRUE;
-    }
-  }
-
-  // Build list of arrays for the permissions.
-  $permissions = array();
-  foreach (array_keys($types) as $type) {
-    $permissions[$type] = user_access('create a ' . $type . ' message instance', $account);
-  }
-
-  return $permissions;
-}
-
-/**
- * Implements hook_menu().
+ * Extend the message entity type's field by providing display handlers.
  */
-function message_ui_menu() {
-  $items = array();
-
-  $items['message/%message'] = array(
-    'title' => 'Viewing a message',
-    'description' => 'Select a message to create an instance.',
-    'page callback' => 'message_ui_show_message',
-    'page arguments' => array(1),
-    'access callback' => 'message_ui_access_control',
-    'access arguments' => array('view', 1),
-  );
-
-  $items['admin/content/message/create'] = array(
-    'title' => 'Create a new message',
-    'description' => 'Select a message to create an instance.',
-    'page callback' => 'message_ui_create_new_message_instance_list',
-    'access callback' => 'message_ui_access_control',
-    'access arguments' => array('create', 'any'),
-    'type' => MENU_LOCAL_ACTION,
-    'weight' => -10,
-  );
-
-  if ($types = message_ui_get_types()) {
-    foreach ($types as $type => $title) {
-      $items['admin/content/message/create/' . str_replace('_', '-', $type)] = array(
-        'title' => ucfirst(str_replace('_', ' ', $title)),
-        'description' => 'Create a new message' . $title . ' instance',
-        'page callback' => 'drupal_get_form',
-        'page arguments' => array('message_ui_instance_message_manage', $type),
-        'access arguments' => array('create', $type),
-        'access callback' => 'message_ui_access_control',
-      );
+function message_ui_entity_base_field_info_alter(&$fields, EntityTypeInterface $entity_type) {
+    // Alter the uid and created field to include display settings.
```

Comment:
@RoySegall It allows the "Authoring Information" or "Owner Information" tab in the vertical tabs to show on the entity form. In node this is in Node.php, but we either have to put this in Message.php or here in an alter hook.